Research/QuantBook.cs

```diff
@@ -95,8 +95,21 @@ public QuantBook() : base()
                     _pandas = Py.Import("pandas");
                 }
 
-                // By default, set start date to end data which is yesterday
-                SetStartDate(EndDate);
+                // Issue #4892 : Set start time relative to NY time 
+                // when the data is available from the previous day
+                var newYorkTime = DateTime.UtcNow.ConvertFromUtc(TimeZones.NewYork);
+                var hourThreshold = Config.GetInt("qb-data-hour", 9);
+                
+                // If it is after our hour threshold; then we can use today 
+                if (newYorkTime.Hour >= hourThreshold)
+                {
+                    SetStartDate(newYorkTime);
+                }
+                else
+                {
+                    SetStartDate(newYorkTime - TimeSpan.FromDays(1));
```

Comment:
Thanks! I think it would be clearer if we kept existing behavior in this `else`, using `EndDate` -> `SetStartDate(EndDate);` but for the `if` above I think we should also set the end date to `newYorkTime`? We might want to double check what are the resulting dates being set for them

Reply:
EndDate is set by base (`QCAlgorithm`) to `DateTime.Now.AddDays(-1)`; 

In the case of EndDate being -1 days from StartDate it doesn't cause any problems with History requests. I think this would only be an issue in the case of QB running as an Algorithm. 

In the `else` statement if we use EndDate it is possible for a user in UTC + 12 (with default 9AM hour threshold) to pass the first `if` and still set the date to same value. I figure this way we cover the edge case and never make requests for data that is not available yet.

Reply:
I just checked setting `EndDate` to newYorkTime, but it actually has a verification check that stops anything greater than yesterday being set :/

```
            if (end > DateTime.Now.Date.AddDays(-1))
            {
                end = DateTime.Now.Date.AddDays(-1);
            }
```

So we won't be able to change it to the current day.

Reply:
Makes sense thank you, let's create a separate issue to fix the end date being off from the algorithm timezone